### PR TITLE
Fixplaneview

### PR DESCRIPTION
**Pull request by pmukherj**

### CODE REVIEW

File: Assets/Placenote/LibPlacenote/LibPlacenote.cs [hunk 1]
```diff
@@ -457,6 +460,16 @@ static void OnPose (ref PNTransformUnity outputPose, ref PNTransformUnity arkitP
 	}
 
 
+	/// <summary>
+	/// Return a transform in the current ARKit frame transformed into the inertial frame w.r.t the current Placenote Map
+	/// </summary>
+	/// <param name="status">Transform in the ARKit frame of reference (e.g: acquired from a screen touch)</param>
+	public Matrix4x4 processPose(Matrix4x4 poseInARKit) {
+		return (mCurrentTransform * poseInARKit);
```
Comment (y22ma) [inline Assets/Placenote/LibPlacenote/LibPlacenote.cs]: We should check whether mCurrentTransform is valid before we try to return a pose.

File: Assets/Placenote/Examples/PlacenoteSampleView.cs [hunk 2]
```diff
@@ -238,32 +250,62 @@ public void OnDeleteMapClicked ()
...
+		mPlaneDetectionToggle.SetActive (true);
+		Debug.Log ("Started Session");
 		LibPlacenote.Instance.StartSession ();
 	}
 
-
+	public void OnTogglePlaneDetection() {
+		planeViz = !planeViz;
+		configureSession (true);
+		Debug.Log ("Plane Viz" + planeViz.ToString ());
```
Comment (y22ma) [inline Assets/Placenote/Examples/PlacenoteSampleView.cs]: add a space or ": " after Plane Viz

File: Assets/Placenote/Examples/PlacenoteSampleView.cs [hunk 3]
```diff
@@ -238,32 +250,62 @@ public void OnDeleteMapClicked ()
...
+		
 	private void StartARKit ()
 	{
 		mLabelText.text = "Initializing ARKit";
 		Application.targetFrameRate = 60;
+		configureSession (false);
+	}
+
+
+	private void configureSession(bool clearPlanes) {
```
Comment (y22ma) [inline Assets/Placenote/Examples/PlacenoteSampleView.cs]: Should be ConfigureSession to keep style consistent

File: Assets/Placenote/Examples/PlacenoteSampleView.cs [hunk 4]
```diff
@@ -50,6 +52,7 @@ public class PlacenoteSampleView : MonoBehaviour, PlacenoteListener
 	private bool mARKitInit = false;
 	private List<ShapeInfo> shapeInfoList = new List<ShapeInfo> ();
 	private List<GameObject> shapeObjList = new List<GameObject> ();
+	private bool planeViz = false;
```
Comment (y22ma) [inline Assets/Placenote/Examples/PlacenoteSampleView.cs]: Doesn't make sense to use a separate boolean variable? Just use the state from the toggle.

File: Assets/Placenote/Examples/PlacenoteSampleView.cs [hunk 5]
```diff
@@ -238,32 +250,62 @@ public void OnDeleteMapClicked ()
...
 		mMappingButtonPanel.SetActive (true);
-
+		mPlaneDetectionToggle.SetActive (true);
+		Debug.Log ("Started Session");
 		LibPlacenote.Instance.StartSession ();
 	}
 
-
+	public void OnTogglePlaneDetection() {
+		planeViz = !planeViz;
```
Comment (y22ma) [inline Assets/Placenote/Examples/PlacenoteSampleView.cs]: see the top comment

File: Assets/Placenote/Examples/PlacenoteSampleView.cs [hunk 6]
```diff
@@ -238,32 +250,62 @@ public void OnDeleteMapClicked ()
...
+		configureSession (false);
+	}
+
+
+	private void configureSession(bool clearPlanes) {
 		ARKitWorldTrackingSessionConfiguration config = new ARKitWorldTrackingSessionConfiguration ();
-		config.planeDetection = UnityARPlaneDetection.Horizontal;
+
+
+		if (planeViz) {
```
Comment (y22ma) [inline Assets/Placenote/Examples/PlacenoteSampleView.cs]: see the top comment

**Review by y22ma — CHANGES_REQUESTED** [review]

File: Assets/Placenote/LibPlacenote/LibPlacenote.cs (hunk 1 above)
Comment (pmukherj) [reply Assets/Placenote/LibPlacenote/LibPlacenote.cs]: in native iOS, i just warn the user that I'm just returning the same transform back. Same solution here good? 
https://github.com/Placenote/PlacenoteSDK-iOS/blob/fixplaneview/Pods/PlacenoteSDK/LibPlacenote.swift#L328

Comment (y22ma) [reply Assets/Placenote/LibPlacenote/LibPlacenote.cs]: I don't know I feel like that's not a strong enough signal. There's no way for user to programmatically react to it. Returning a null or throwing an exception makes more sense to me.

File: Assets/Placenote/Examples/PlacenoteSampleView.cs (hunk 2 above)
Comment (pmukherj) [reply Assets/Placenote/Examples/PlacenoteSampleView.cs]: woops debug msg, no need for it to stay. removing.

**Review by nmathew87 — COMMENTED** [review]
Looks good

File: Assets/Placenote/LibPlacenote/LibPlacenote.cs [hunk 7]
```diff
@@ -457,6 +460,17 @@ static void OnPose (ref PNTransformUnity outputPose, ref PNTransformUnity arkitP
 	}
 
 
+	/// <summary>
+	/// Return a transform in the current ARKit frame transformed into the inertial frame w.r.t the current Placenote Map. Will return null 
+	/// mapping is not initialized 
+	/// </summary>
+	/// <param name="status">Transform in the ARKit frame of reference (e.g: acquired from a screen touch)</param>
+	public Matrix4x4? ProcessPose(Matrix4x4 poseInARKit) {
+		return (mCurrentTransform * poseInARKit);
```
Comment (y22ma) [inline Assets/Placenote/LibPlacenote/LibPlacenote.cs]: This will throw an exception

**Review by y22ma — CHANGES_REQUESTED** [review]

**Review by y22ma — APPROVED** [review]

**Pull request merged**